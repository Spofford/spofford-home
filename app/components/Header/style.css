.fullHeader {
  height:0px;
}

header.container {
  width:100%;
  height:65px;
  transition: height .5s ease;
  background-color:white;
  z-index:1;
  overflow:hidden;
  box-shadow: 1px 2px #BBBBBB;
}

header h1 {
  color:white;
}

header.invisible {
  height:0px;
}

header.container > div, .drawer-head > div {
  float:left;
  padding:15px;
}

.drawer-head .header-container {
  margin-left:5%;
}

.drawer-head > .search-container.hidden {
  width:20%;
}

header .icon-container input {
  border: 1px solid #e5e5e5;
  border-radius: 2px;
  box-shadow: inset 0px -1px 15px #e5e5e5;
  font-family: 'Montserrat',sans-serif;
  font-size: 24px;
  height: 47px;
  padding: 0 5px 0 55px;
  position: absolute;
  width: 75%;
  top: 9px;
}

header .fa-search {
  z-index: 2;
  position: absolute;
  left: 20px;
  color:black;
}

header .cancelTrigger {
  position: absolute;
  right: 0;
  top: 22px;
}

header .icon-container input:focus {
  border-color:#979797;
  outline:none;
}

header .search-container {
  position:absolute;
  width:95%;
}

header .search-container.hidden {
  position:absolute;
  width:20%;
  right:0;
}

header .search-container.hidden .link-container {
  display:block;
}

header .search-container .link-container {
  display:none;
  float:right;
  margin-right:70px;
  padding-top: 3px;
}

header .search-container .link-container a, header .search-container .link-container span {
  font-size: 1.5em;
  margin-top: 3px;
  text-transform: uppercase;
  text-decoration: none;
  color:#333;
}

.search-container.hidden input, .search-container.hidden .cancelTrigger {
  display:none;
}

header.container > .header-container {
  width:70%;
}

.drawer > .header-container {
  width:calc(80%-60px);
}

header .header-container h1, .drawer h2 {
  text-align:center;
}

header h1, .header-container > h1 {
    letter-spacing: 5px;
    font-size:1.5em;
}

header > .icon-container .fa-search {
  float:right;
}

.fullHeader nav a {
  text-decoration: none;
}

.fullHeader nav li {
  text-transform: uppercase;
}

.fullHeader .address {
  font-size:1em;
}

.icon-container {
  cursor:pointer;
}

.icon-container span.menu-label {
  font-size: 1.5em;
  text-transform: uppercase;
  margin-left:20px;
  margin-top:3px;
  display:none;
}

.fa-times, .fa-bars {
  cursor: pointer;
}

.fa-bars {
  float:left;
}

.drawer-head {
  width:100%;
  float:left;
}

.nav-container {
  height:0px;
  position:fixed;
  top:0;
  z-index:3;
}

.nav-container.visible /*, .nav-container.visible .overlay*/ {
  width:100%;
}

.nav-container.visible .drawer {
  transform: translateX(0px);
}

.nav-container .drawer {
  width:380px;
  background:white;
  height:100%;
  position:fixed;
  z-index:3;
  transform: translateX(-390px);
  transition: 1s ease-in-out;
}

.nav-container.visible .overlay {
  height:100%;
  width:100%;
  opacity:.5;
  transition: opacity 1s linear;
}

.nav-container .overlay {
  position:fixed;
  height:0px;
  opacity:0;
  width:0px;
  z-index:2;
  transition: height 0s 1s, width 0s 1s, opacity 1s linear;
  background:#585858;
}



/*
.nav-container.visible .overlay {
  background: #585858;
  opacity:.5;
  transition: opacity 2s linear;
}
*/

.drawer li {
  font-size:1.2em;
  line-height:2em;
}

.drawer nav {
  padding: 0px 10px;
}

.drawer hr {
  margin: 15px 0px;
}

.drawer i {
  margin-right:10px;
}

@media (min-width: 700px) {
  .icon-container span.menu-label {
    display:block;
    float:left;
  }
  .search-container.hidden .fa-search {
    left:auto;
    right:20px;
  }
  .search-container {
    max-width: 75%;
    right: 10px;
  }
  header .cancelTrigger {
    right: 10;
  }
}
