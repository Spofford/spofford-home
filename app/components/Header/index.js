import React from "react";
import FontAwesome from "react-fontawesome";
import cssModules from 'react-css-modules';
import classNames from 'classnames';
import style from "./style.css";
import { Link, Redirect } from 'react-router-dom'
import * as contentful from 'contentful'
import { connect } from "react-redux"
import Actions from "../../redux/actions"

const client = contentful.createClient({
  space: 'cahjy08ew1qz',
  accessToken: '37c6ec31a1a6cb3f533f51fa4c4af8fee88e2f910d9879eb79b2d073ae8cc499'
})

export class Header extends React.Component {

  constructor() {
    super()

    this.state = {
      scrolled: false,
      lastScrollTop: 0,
      isToggleOn: false,
      searchHidden:true,
      inputValue:'',
      redirect: false
    }

    this.handleScroll = this.handleScroll.bind(this);
    this.toggleDrawer = this.toggleDrawer.bind(this);
    this.showSearch = this.showSearch.bind(this);
    this.hideSearch = this.hideSearch.bind(this);
    this.returnSearch = this.returnSearch.bind(this)
    this.signOut = this.signOut.bind(this)
  }

  renderRedirect = () => {
    if (this.state.redirect) {
      return <Redirect to={'/search/' + this.state.inputValue} />
    }
  }

  componentDidMount() {

  }

  signOut() {
    this.props.dispatch(Actions.logout())
  }


  handleScroll(event) {
    var st = window.pageYOffset || document.documentElement.scrollTop;

    if (st > this.state.lastScrollTop){
      this.setState({'scrolled': true});
    } else {
      this.setState({'scrolled': false});
    }
    if (st <= 0) {
      this.setState({'lastScrollTop': 0})
    } else {
      this.setState({'lastScrollTop': st})
    }
  }

  toggleDrawer() {
    this.setState(prevState => ({
      isToggleOn: !prevState.isToggleOn
    }));
    var elem = document.getElementsByTagName("body")[0];
    if (elem.style.overflow=="hidden") {
      elem.style.overflow="auto"
    } else {
      elem.style.overflow="hidden"
    }
  }

  showSearch() {
    this.setState({
      searchHidden: false
    });
  }

  hideSearch() {
    this.setState({
      searchHidden: true,
      inputValue:''
    });
  }

  returnSearch(evt) {
    if(evt.keyCode == 13) {
      this.setState({
        redirect: true
      })
    }
  }

  updateInputValue(evt) {
    this.setState({
      inputValue: evt.target.value
    })
  }

  headerState() {
    if (this.props.user.email) {
      return <span onClick={this.signOut}>Sign Out</span>
    } else {
      return <Link to="/login">Login</Link>
    }

  }

  render() {
    console.log(this.props.user)

    var liClasses = classNames({
      'container': true,
      'invisible': this.state.scrolled
    });

    var navClasses = classNames({
      'nav-container': true,
      'visible': this.state.isToggleOn
    });

    var searchClasses = classNames(
      'icon-container', 'search-container',
      {'hidden': this.state.searchHidden}
    )

    return (
      <div className='fullHeader'>
        <header className={liClasses}>
          <div className='icon-container' onClick={this.toggleDrawer}><FontAwesome name='bars' size='2x' /><span className="menu-label">Menu</span></div>
          <div className='header-container'><h1>SPOFFORD</h1></div>
          <div className={searchClasses}>
            <div className='link-container'>
              {this.headerState()}
            </div>
            <FontAwesome onClick={this.showSearch} name='search' size='2x' />
            {this.renderRedirect()}
            <input placeholder="Search" value={this.state.inputValue} className="search-field" onKeyDown={this.returnSearch} tabIndex="0" onChange={evt => this.updateInputValue(evt)} />
            <span onClick={this.hideSearch} className='cancelTrigger'>Cancel</span>
          </div>
        </header>
        <div className={navClasses}>
          <div className='drawer'>
            <div className='drawer-head'>
              <div className='icon-container' onClick={this.toggleDrawer}><FontAwesome name='times' size='2x' /></div>
              <div className='header-container'><h1>SPOFFORD</h1></div>
            </div>
            <nav>
              <ul>
                <li><Link to="/" onClick={this.toggleDrawer}>Home</Link></li>
                <li>Collections&mdash;Coming Soon</li>
                <li><Link to="/about" onClick={this.toggleDrawer}>About</Link></li>
                <li><Link to="/studios" onClick={this.toggleDrawer}>Studios</Link></li>
                <li>Society&mdash;Coming Soon</li>
                <hr />
                <li onClick={this.toggleDrawer}><a href="https://share.hsforms.com/10eEyKMOWQzi742fNhPUwZQ2emyf" target="_blank">Newsletter&mdash;Sign Up</a></li>
                <hr />
                <a target="_blank" href="https://www.facebook.com/spofforddesign/"><i className="fab fa-facebook fa-3x"></i></a>
                <a target="_blank" href="https://www.instagram.com/spofforddesign/"><i className="fab fa-instagram fa-3x"></i></a>
                <a target="_blank" href="https://twitter.com/spofforddesign"><i className="fab fa-twitter fa-3x"></i></a>
                <a target="_blank" href="https://www.pinterest.com/spofforddesign"><i className="fab fa-pinterest fa-3x"></i></a>
                <hr />
                <li className="address">info&#64;spofforddesign.com</li>
              </ul>
            </nav>
          </div>
          <div className='overlay'></div>
        </div>
      </div>
    )
  }
};

const mapStateToProps = state => ({
  user: state.user
})

export default connect(mapStateToProps)(cssModules(Header, style))
