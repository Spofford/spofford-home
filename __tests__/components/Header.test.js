import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import configureStore from 'redux-mock-store';

import Header from '../../app/components/Header';

const mockStore = configureStore();
const initialState = {
  user: {
    email: "amhasler@gmail.com",
    first_name: "Adam",
    last_name: "Hasler",
    role: 1,
    id: 20,
    charges: []
  }
}
const store = mockStore(initialState);

describe('<Header />', () => {
  describe('render()', () => {
    test('renders the component', () => {
      const wrapper = shallow(<Header store={store} />);
      const component = wrapper.dive();

      expect(toJson(component)).toMatchSnapshot();
    });
  });
});
