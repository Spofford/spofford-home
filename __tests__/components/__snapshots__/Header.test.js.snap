// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> render() renders the component 1`] = `
<div
  className="fullHeader"
>
  <header
    className="container"
  >
    <div
      className="icon-container"
      onClick={[Function]}
    >
      <FontAwesome
        id="open-toggle"
        name="bars"
        size="2x"
      />
      <span
        className="menu-label"
      >
        Menu
      </span>
    </div>
    <div
      className="header-container"
    >
      <h1>
        SPOFFORD
      </h1>
    </div>
    <div
      className="search-container hidden"
    >
      <FontAwesome
        name="search"
        onClick={[Function]}
        size="2x"
      />
      <input
        className="search-field"
        onChange={[Function]}
        onKeyDown={[Function]}
        placeholder="Search"
        tabIndex="0"
        value=""
      />
      <span
        className="cancelTrigger"
        onClick={[Function]}
      >
        Cancel
      </span>
    </div>
  </header>
  <div
    className="nav-container"
  >
    <div
      className="drawer"
      id="header-drawer"
    >
      <div
        className="drawer-head"
      >
        <div
          className="icon-container"
          onClick={[Function]}
        >
          <FontAwesome
            name="times"
            size="2x"
          />
        </div>
        <div
          className="header-container"
        >
          <h1>
            SPOFFORD
          </h1>
        </div>
      </div>
      <nav>
        <ul>
          <li>
            <Link
              onClick={[Function]}
              replace={false}
              to="/"
            >
              Home
            </Link>
          </li>
          <li>
            Collections—Coming Soon
          </li>
          <li>
            <Link
              onClick={[Function]}
              replace={false}
              to="/about"
            >
              About
            </Link>
          </li>
          <li>
            <Link
              onClick={[Function]}
              replace={false}
              to="/studios"
            >
              Studios
            </Link>
          </li>
          <li>
            Society—Coming Soon
          </li>
          <li>
            <Link
              onClick={[Function]}
              replace={false}
              to="/show"
            >
              2019 Group Show
            </Link>
          </li>
          <hr />
          <li
            onClick={[Function]}
          >
            <a
              href="https://share.hsforms.com/10eEyKMOWQzi742fNhPUwZQ2emyf"
              target="_blank"
            >
              Newsletter—Sign Up
            </a>
          </li>
          <li
            className="login-toggle"
            onClick={[Function]}
          >
            <span
              onClick={[Function]}
            >
              Sign Out
            </span>
          </li>
          <hr />
          <a
            href="https://www.facebook.com/spofforddesign/"
            target="_blank"
          >
            <i
              className="fab fa-facebook fa-3x"
            />
          </a>
          <a
            href="https://www.instagram.com/spofforddesign/"
            target="_blank"
          >
            <i
              className="fab fa-instagram fa-3x"
            />
          </a>
          <a
            href="https://twitter.com/spofforddesign"
            target="_blank"
          >
            <i
              className="fab fa-twitter fa-3x"
            />
          </a>
          <a
            href="https://www.pinterest.com/spofforddesign"
            target="_blank"
          >
            <i
              className="fab fa-pinterest fa-3x"
            />
          </a>
          <hr />
          <li
            className="address"
          >
            info@spofforddesign.com
          </li>
        </ul>
      </nav>
    </div>
    <div
      className="overlay"
    />
  </div>
</div>
`;
